### PR TITLE
Updated and fixed bugs in pubmlst and ncbi downloads

### PR DESCRIPTION
**Pull request by talnor**
# Description

_The features of this PR primarily concerns bioinformaticians_

_Summary of the changes made:_
* Bug fix for ncbi reference download
* Updated pubmlst external reference download
* Added force_update flag for analyse command

_If not self-evident, mention what prompted the change._
The pubmlst website was updated and those mlst profiles that were only linked to pubmlst could therefore not be updated. 

## Primary function of PR
- [ ] Hotfix
- [X] Patch
- [ ] Minor functionality improvement
- [ ] New type of analysis
- [ ] Backward-breaking functionality improvement
- [ ] This change requires internal documents to be updated
- [ ] This change requires another repository to be updated

# Testing

_Test routine to verify the stability of the PR:_
- _`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh fix_referencer`_
- _`us`_
- _`source activate S_microSALT`_

- _`cg workflow microsalt config-case -t 244354`_
- _`Remove the following references from the stage genomes directory: NC_016845.1  and NC_011751.1`_
- _`microSALT analyse /home/proj/stage/microbial/queries/244354.json --input /home/proj/stage/microbial/fastq/ACC6369`_
- _`Verify that the removed references are redownloaded and indexed`_
- _`Verify that there are no unexpected errors when running`_

- _`cg workflow microsalt config-case ACC6369A41`_
- _`microSALT analyse /home/proj/stage/microbial/queries/ACC6369A42.json --input /home/proj/stage/microbial/fastq/ACC6369/ACC6369A41  --force_update`_
- _`Verify that the MLST profiles for E. coli are updated`_
- _`Verify that there are no unexpected errors when running`_

## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._
References and MLST updated, see below for results. Also no errors from force flag. 

# Sign-offs
- [x] Code tested by @talnor 
- [x] Approved to run at Clinical-Genomics by @sylvinite

### CODE REVIEW

**Comment by talnor** [comment]
Fixes issue #122

File: microSALT/cli.py [hunk 1]
```diff
@@ -120,6 +120,9 @@ def root(ctx):
 @click.option(
     "--skip_update", default=False, help="Skips downloading of references", is_flag=True
 )
+@click.option(
+    "--force_update", default=False, help="Forces downloading of pubMLST references", is_flag=True
```
Comment (sylvinite) [inline microSALT/cli.py:124]: Skip and force could be on/off for the same flag.

File: microSALT/utils/referencer.py [hunk 2]
```diff
@@ -135,63 +135,55 @@ def index_db(self, full_dir, suffix):
...
-                # In order to get ecoli #1
+            query = urllib.request.urlopen(url).read()
+            root = ET.fromstring(query)
+            for entry in root:
+                # Check organism
+                species = entry.text
+                organ = species.strip().lower().replace(" ", "_") 
                 if "escherichia_coli" in organ and "#1" in organ:
                     organ = organ[:-2]
+                if organ not in self.organisms:
```
Comment (sylvinite) [inline microSALT/utils/referencer.py:148]: Placement of not like this makes it invalid. Effectively it now says "if organ in self.organisms"

File: microSALT/cli.py (hunk 1 above)
Comment (sylvinite) [reply microSALT/cli.py:124]: Nevermind. 
Doing that means the default value for the flag has to be NULL. Which will cause confusion.

**Review by sylvinite — CHANGES_REQUESTED** [review]

**Review by sylvinite — APPROVED** [review]

**Comment by talnor** [comment]
Reference download:
INFO - Downloaded reference NC_011751.1
INFO - Downloaded reference NC_016845.1

[21|278|284] 13d [hiseq.clinical@hasta:/home/proj/stage/microbial] [S_microSALT] $ ll references/genomes/NC_016845.1*
-rw-rw-r--+ 1 hiseq.clinical hasta-development 5410232 Nov 11 11:09 references/genomes/NC_016845.1.fasta
-rw-rw-r--+ 1 hiseq.clinical hasta-development      24 Nov 11 11:09 references/genomes/NC_016845.1.fasta.amb
-rw-rw-r--+ 1 hiseq.clinical hasta-development     115 Nov 11 11:09 references/genomes/NC_016845.1.fasta.ann
-rw-rw-r--+ 1 hiseq.clinical hasta-development 5334020 Nov 11 11:09 references/genomes/NC_016845.1.fasta.bwt
-rw-rw-r--+ 1 hiseq.clinical hasta-development      29 Nov 11 11:09 references/genomes/NC_016845.1.fasta.fai
-rw-rw-r--+ 1 hiseq.clinical hasta-development 1333487 Nov 11 11:09 references/genomes/NC_016845.1.fasta.pac
-rw-rw-r--+ 1 hiseq.clinical hasta-development 2667024 Nov 11 11:09 references/genomes/NC_016845.1.fasta.sa
[19|276|288] θ60° 13d [hiseq.clinical@hasta:/home/proj/stage/microbial] [S_microSALT] $ ll references/genomes/NC_011751.1*
-rw-rw-r--+ 1 hiseq.clinical hasta-development 5276461 Nov 11 11:09 references/genomes/NC_011751.1.fasta
-rw-rw-r--+ 1 hiseq.clinical hasta-development      12 Nov 11 11:09 references/genomes/NC_011751.1.fasta.amb
-rw-rw-r--+ 1 hiseq.clinical hasta-development      80 Nov 11 11:09 references/genomes/NC_011751.1.fasta.ann
-rw-rw-r--+ 1 hiseq.clinical hasta-development 5202176 Nov 11 11:09 references/genomes/NC_011751.1.fasta.bwt
-rw-rw-r--+ 1 hiseq.clinical hasta-development      29 Nov 11 11:09 references/genomes/NC_011751.1.fasta.fai
-rw-rw-r--+ 1 hiseq.clinical hasta-development 1300524 Nov 11 11:09 references/genomes/NC_011751.1.fasta.pac
-rw-rw-r--+ 1 hiseq.clinical hasta-development 2601096 Nov 11 11:09 references/genomes/NC_011751.1.fasta.sa

**Comment by talnor** [comment]
Update external MLST:
```

INFO - Downloading new MLST profiles for Escherichia coli#1

INFO - Re-indexed contents of /home/proj/stage/microbial/references/ST_loci/escherichia_coli
```
```
references/ST_loci/escherichia_coli:
total 11776
-rw-rw-r--+ 1 hiseq.clinical hasta-development  43130 Nov 11 16:20 adk.nhr
-rw-rw-r--+ 1 hiseq.clinical hasta-development  11384 Nov 11 16:20 adk.nin
-rw-rw-r--+ 1 hiseq.clinical hasta-development   3780 Nov 11 16:20 adk.nog
-rw-rw-r--+ 1 hiseq.clinical hasta-development  26072 Nov 11 16:20 adk.nsd
-rw-rw-r--+ 1 hiseq.clinical hasta-development    702 Nov 11 16:20 adk.nsi
-rw-rw-r--+ 1 hiseq.clinical hasta-development 126496 Nov 11 16:20 adk.nsq
-rw-rw-r--+ 1 hiseq.clinical hasta-development 511630 Nov 11 16:20 adk.tfa
-rw-rw-r--+ 1 hiseq.clinical hasta-development  70558 Nov 11 16:20 fumC.nhr
-rw-rw-r--+ 1 hiseq.clinical hasta-development  18032 Nov 11 16:20 fumC.nin
-rw-rw-r--+ 1 hiseq.clinical hasta-development   5996 Nov 11 16:20 fumC.nog
-rw-rw-r--+ 1 hiseq.clinical hasta-development  46454 Nov 11 16:20 fumC.nsd
```

**Pull request merged**